### ISSUE TITLE
allocator<T>::allocate(size_t n) 'n' exceeds maximum supported size

### ISSUE THREAD

**Issue by tobrun** [mapbox/mapbox-gl-native#8275]
```
 Process: com.mapbox.mapboxsdk.testapp, PID: 15186
 java.lang.Error: allocator<T>::allocate(size_t n) 'n' exceeds maximum supported size
 at com.mapbox.mapboxsdk.maps.NativeMapView.nativeRender(Native Method)
 at com.mapbox.mapboxsdk.maps.NativeMapView.render(NativeMapView.java:174)
 at com.mapbox.mapboxsdk.maps.MapView.onDraw(MapView.java:420)
 at android.view.View.draw(View.java:17185)
 at android.view.View.updateDisplayListIfDirty(View.java:16167)
 at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3711)
 at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3691)
 at android.view.View.updateDisplayListIfDirty(View.java:16130)
 at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3711)
 at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3691)
 at android.view.View.updateDisplayListIfDirty(View.java:16130)
 at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3711)
 at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3691)
 at android.view.View.updateDisplayListIfDirty(View.java:16130)
 at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3711)
 at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3691)
 at android.view.View.updateDisplayListIfDirty(View.java:16130)
 at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3711)
 at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3691)
 at android.view.View.updateDisplayListIfDirty(View.java:16130)
 at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3711)
 at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3691)
 at android.view.View.updateDisplayListIfDirty(View.java:16130)
 at android.view.ThreadedRenderer.updateViewTreeDisplayList(ThreadedRenderer.java:648)
 at android.view.ThreadedRenderer.updateRootDisplayList(ThreadedRenderer.java:654)
 at android.view.ThreadedRenderer.draw(ThreadedRenderer.java:762)
 at android.view.ViewRootImpl.draw(ViewRootImpl.java:2800)
 at android.view.ViewRootImpl.performDraw(ViewRootImpl.java:2608)
 at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2215)
 at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1254)
 at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6337)
 at android.view.Choreographer$CallbackRecord.run(Choreographer.java:874)
 at android.view.Choreographer.doCallbacks(Choreographer.java:686)
 at android.view.Choreographer.doFrame(Choreographer.java:621)
 at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:860)
 at android.os.Handler.handleCallback(Handler.java:751)
 at android.os.Handler.dispatchMessage(Handler.java:95)
 at android.os.Looper.loop(Looper.java:154)
 at android.app.ActivityThread.main(ActivityThread.java:6119)
 at java.lang.reflect.Method.invoke(Native Method)
 at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
 at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
```

![ezgif com-video-to-gif 15](https://cloud.githubusercontent.com/assets/2151639/23568791/356a550c-002a-11e7-990d-8f3f91d07c01.gif)

Similar use-case as in #8273, but different crash log.

**Comment by kingfisherphuoc** [comment]
@tobrun I also got this error when trying to use method `mapboxMap.getMarkers()` many times.

**Comment by jfirebaugh** [comment]
We'll need to get the native stack trace for this somehow.

**Comment by jfirebaugh** [comment]
This was probably fixed by #8298.
